### PR TITLE
Windows CI: test-unit on pkg\archive part 2

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

Part 2 of porting the unit-tests for pkg\archive to Windows CI (part 1 @ https://github.com/docker/docker/pull/20284). There's a lot of tests that now pass, and several annotations in there as for further work to do.

### CODE REVIEW

**Comment by calavera** [comment]
I think we should stop using `/temp` and `Getenv("TEMT")` altogether. Use `ioutil.TempDir` should do the right thing in either OS:

https://golang.org/pkg/io/ioutil/#TempDir

**Comment by jhowardmsft** [comment]
@calavera Generally, yes, I agree. However, in the particular case you're citing here, it's actually going to complicate the porting of the tests. The reason being that as some things are being through APIs which require "real" (c:\foo\bar) Windows paths, and others through bash commands which requires "fake" (/c/foo/bar) Windows paths. Golang has no such capability built in. Sure, it's an achievable goal, but in the context of the actual changes being made here, IMO the extra work right now doesn't seem worth it yet. Certainly as a potential follow-up PR though.

File: pkg/archive/changes_test.go [hunk 1]
```diff
@@ -341,6 +362,11 @@ func mutateSampleDir(t *testing.T, root string) {
 }
 
 func TestChangesDirsMutated(t *testing.T) {
+	// TODO Windows. There may be a way of running this, but turning off for now
```
Comment (thaJeztah) [inline pkg/archive/changes_test.go]: Should we have a tracker issue tracking all these todo's?

**Comment by jhowardmsft** [comment]
You're going to have hundreds! Grep the code for 'TODO Windows'... Honestly think it would be defeatist to have issues for all of them.

**Comment by cyphar** [comment]
@jhowardmsft Not an issue for each one, a tracking issue for all of them (a checklist of things we have to look at).

**Comment by jhowardmsft** [comment]
Sure. Feel free to open an issue. But I really don't see the need when it could (will) take many, many months and multiple releases to get resolved. cf., it's not like there's an issue open for every matching "FIXME:" and "TODO" on the Linux side....

**Comment by thaJeztah** [comment]
@jhowardmsft don't read too much in my comment, I just noticed there's a lot of TODO's added, and was wondering if there's a desire to have a tracker issue (using the "checkbox" notation) to track progress.

**Comment by calavera** [comment]
LGTM

File: pkg/archive/changes_test.go (hunk 1 above)
Comment (cpuguy83) [inline pkg/archive/changes_test.go]: `git grep 'TODO Windows'` :)

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**